### PR TITLE
Put global ts definitions in separate file

### LINKED ISSUE #29304: Big.js in global scope prone to causing errors when used in Node
If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @nycdotnet @googol 

because this declares the Big.js object in the global scope, when using code modularly in Node/in frontend mobile/browser codebases using webpack, where the library is not present in the global scope, the typings say that it always is; this has caused me to introduce bugs multiple times now thinking that the code is fine, but it's not.

I'm not sure how other libraries like `jquery` or what not handle this issue, or if they just have the same problems.

### PR DESCRIPTION
**Pull request by ricmello**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Put global ts definitions in a separate global.d.ts file because it can break projects that haven't imported Big.js. TS will not throw errors if Big.js is at global scope.

Closes #29304

### CODE REVIEW

File: types/big.js/index.global.d.ts
Comment (sandersn) [inline types/big.js/index.global.d.ts]: does this change anything about the tests? Do the tests already cover global and module usage?

**Review by sandersn — CHANGES_REQUESTED** [review]
For a package this popular, I'd really like @nycdotnet and @googol to have a look before merging it. I don't understand the intent myself; I think it would help to see tests for the global definitions so I can compare to the tests for the module.

**Comment by osdiab** [comment]
Hm just got bitten by this issue. Any chance this can be reconsidered?

**Comment by sandersn** [comment]
Sure, at this point I'd recommend making a new PR based on this one. It seems like the hard part will be contacting the maintainers to get a review.

**Comment by nycdotnet** [comment]
Sorry - I suppose it's fine to do this if this is what people are doing nowadays for hybrid global/module packages.

**Comment by nycdotnet** [comment]
Ping me on a new PR and I'll approve it if looks good.

**Comment by ricmello** [comment]
Sure! I can do it

**Pull request closed without merging**
